File: Dia02/d2p2/src/main.js
readFileSync = require('fs').readFileSync;

let data = readFileSync('../../input.txt', 'utf8').split('\n')//.map((v, i, arr) => v.map((v, i, arr) => parseInt(v.trim()));
let d = 0;
let f = 0;
let aim = 0;
for (let line of data) {
	let s = line.split(' ');

	if(s[0] === 'up') {
		aim -= parseInt(s[1])
	}else if(s[0] === 'down') {
		aim += parseInt(s[1])
	}else{
		let x = parseInt(s[1])
		f += x
		d += x * aim
	}
}

console.log(d*f);

File: Dia02/d2p2/src/Main.java
import java.io.FileInputStream;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		try {
			// the file to be opened for reading
			FileInputStream fis = new FileInputStream("../../input.txt");
			Scanner sc = new Scanner(fis); // file to be scanned
			int d = 0;
			int f = 0;
			int aim = 0;
			// returns true if there is another line to read
			while (sc.hasNext()) {
				String cmd = sc.next();
				int arg = sc.nextInt();
				if (cmd.equals("up")) {
					aim -= arg;
				}else if (cmd.equals("down")) {
					aim += arg;
				}else {
					f += arg;
					d += arg * aim;
				}

			}
			System.out.println(d*f);
			sc.close(); // closes the scanner
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}